src/app/wallet/proposals/proposal-details/proposal-details.component.ts

```diff
@@ -0,0 +1,152 @@
+import { Component, Input, OnInit, OnDestroy } from '@angular/core';
+import { MatDialog } from '@angular/material';
+import * as d3 from 'd3';
+import { Log } from 'ng2-logger';
+
+import {
+  ProposalVoteConfirmationComponent
+} from 'app/modals/proposal-vote-confirmation/proposal-vote-confirmation.component';
+import { ProposalsService } from 'app/wallet/proposals/proposals.service';
+import { SnackbarService } from 'app/core/snackbar/snackbar.service';
+import { VoteOption } from 'app/wallet/proposals/models/vote-option.model';
+import { ModalsHelperService } from 'app/modals/modals-helper.service';
+import { Proposal } from 'app/wallet/proposals/models/proposal.model';
+import { ProposalResult } from 'app/wallet/proposals/models/proposal-result.model';
+import { GraphOption } from 'app/wallet/proposals/models/proposal-result-graph-option.model';
+import { VoteDetails } from 'app/wallet/proposals/models/vote-details.model';
+
+@Component({
+  selector: 'app-proposal-details',
+  templateUrl: './proposal-details.component.html',
+  styleUrls: [
+    './proposal-details.component.scss',
+    '../../../../../node_modules/nvd3/build/nv.d3.css'
+  ]
+})
+export class ProposalDetailsComponent implements OnInit, OnDestroy {
+  log: any = Log.create('proposal.component');
+  @Input() proposal: Proposal;
+  @Input() selectedTab: string;
+  @Input() currentBlockCount: number;
+
+  // pie chart config(s).
+  public graphOptions: any = {
+    chart: {
+      type: 'pieChart',
+      height: 250,
+      width: 250,
+      x: (d) => { return d.description; },
+      y: (d) => { return d.voters; },
+      showLabels: false,
+      donut: true,
+      legend: {
+        margin: {
+          top: 5,
+          right: 35,
+          bottom: 5,
+          left: 0
+        }
+      },
+      color: ['#02E8B0', '#ec4b50', '#108cda', '#f1cc00', '#7e6c95'], // green, red, blue, yellow, violet
+      tooltip: {
+        enabled: true,
+        hideDelay: 0,
+        useInteractiveGuideline: false
+      }
+    }
+  };
+
+  public selectedOption: VoteOption;
+  public proposalResult: ProposalResult;
+  // @TODO create exact type.
+  public voteDetails: VoteDetails;
+  public aleradyVoted: boolean = false
+  destroyed: boolean = false;
+
+  constructor(
+    private dialog: MatDialog,
+    private proposalService: ProposalsService,
+    private snackbarService: SnackbarService,
+    private modelsService: ModalsHelperService
+  ) { }
+
+  ngOnInit() {
+    if (this.proposal) {
+      this.getProposalResult();
+    }
+  }
+
+  getVoteDetails(): void {
+    this.proposalService.get(this.proposal.hash)
+    .takeWhile(() => !this.destroyed)
+    .subscribe((voteDetail: VoteDetails) => {
+      this.voteDetails = voteDetail;
+      this.aleradyVoted = true;
+    }, (error) => {
+      this.log.d('you not yet?', error)
```

Comment:
What does this error message refer to? Can this be re-worded to be somewhat more informative of what went wrong. If the error itself that is being included in the message is sufficiently descriptive, then perhaps just leave the error. Otherwise, perhaps a better prefix indication of the error message would be better.